test/hook_widget_test.dart

```diff
@@ -50,11 +52,50 @@ void main() {
     reset(build);
     reset(didBuild);
     reset(dispose);
+    reset(deactivate);
     reset(initHook);
     reset(didUpdateHook);
     reset(reassemble);
   });
 
+  testWidgets('should call deactivate when removed from and inserted into another place', (tester) async {
+    final _key = GlobalKey();
+    final state = ValueNotifier(false);
+    await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.rtl,
+          child: ValueListenableBuilder(
+              valueListenable: state,
+              builder: (context, bool value, _) => Stack(
+                  children: [
+                    HookBuilder(
+                        key: value ? null : _key,
+                        builder: (context) {
+                          Hook.use(createHook());
+                          return Container();
+                        },
+                    ),
+                    HookBuilder(
+                      key: !value ? null : _key,
+                      builder: (context) {
+                        Hook.use(createHook());
+                        return Container();
+                      },
+                    ),
+                  ]
+              )
+          ),
+        )
+    );
+    await tester.pump();
+    verifyNever(deactivate());
+    state.value = true;
+    await tester.pump();
+    verify(deactivate()).called(1);
```

Comment:
Could you expand it slightly to assert that it calls deactivate on all hooks in order?

And another check that if one deactivate call fails, the others are still called

Reply:
Also it may actually not be needed to use GlobalKey here, since `deactivate` is also called right before `dispose`